src/config/api/api.types.ts

```diff
@@ -0,0 +1,24 @@
+export interface RequestParams<T> {
+  searchText?: string;
+  page?: number;
+  pageSize?: number;
+  sort?: string;
+  filter?: T;
+  with?: string | string[];
```

Comment:
Do we want to promote using `with` for loading relationships from the frontend? I know it's a feature that exists in our backends, but I remember that at some point we were discussing this is a bad practice

Reply:
+1

Reply:
How else would you load relationships instead if you don't promote using the feature?

Reply:
You just don't load them from the frontend, you define the relationships that are returned by an endpoint in the backend, and if you need different versions you can create different endpoints.

It's a matter of who's responsible for defining the response of an endpoint.